### PR TITLE
SALTO-825: added support for skip list for data instances

### PR DESCRIPTION
**Pull request by alonstern**
This PR is to make the new Netsuite instances to support the skip list and partial fetch. 

The changes in the element index are to make the references work when using partial fetch

---
_Release Notes_: 
None

### CODE REVIEW

File: packages/netsuite-adapter/src/filters/data_instances_references.ts [hunk 1]
```diff
@@ -58,10 +59,17 @@ const replaceReference: (
...
-    const dataInstancesMap = await awu(instances.filter(e => e.value.internalId !== undefined))
-      .keyBy(async e => `${(await e.getType()).elemID.name}-${e.value.internalId}`)
+    const dataInstancesMap: Record<string, { elemID: ElemID }> = isPartial ? _.clone(
+      (await elementsSourceIndex.getIndexes()).internalIdsIndex
+    ) : {}
+
+    _.assign(
+      dataInstancesMap,
+      await awu(instances.filter(e => e.value.internalId !== undefined))
+        .keyBy(async e => `${(await e.getType()).elemID.name}-${e.value.internalId}`)
```
Comment (omrilit) [inline packages/netsuite-adapter/src/filters/data_instances_references.ts:71]: super nits:
1. rename e to inst or something
2. I think that we have a util func for `${(await e.getType()).elemID.name}-${e.value.internalId}`

File: packages/netsuite-adapter/src/query.ts [hunk 2]
```diff
@@ -87,7 +92,8 @@ export const andQuery = (firstQuery: NetsuiteQuery, secondQuery: NetsuiteQuery):
 })
 
 export const notQuery = (query: NetsuiteQuery): NetsuiteQuery => ({
-  isTypeMatch: () => true,
+  isTypeMatch: typeName => !query.areAllObjectsMatch(typeName),
+  areAllObjectsMatch: typeName => !query.isTypeMatch(typeName),
```
Comment (omrilit) [inline packages/netsuite-adapter/src/query.ts:96]: I assume that it indicates this area is not covered well with UTs

```suggestion
  isTypeMatch: typeName => !query.isTypeMatch(typeName),
  areAllObjectsMatch: typeName => !query.areAllObjectsMatch(typeName),
```

**Review by omrilit — APPROVED** [review]
Looks great!
Only one important comment.
Approving to save time

File: packages/netsuite-adapter/src/query.ts (hunk 2 above)
Comment (alonstern) [reply packages/netsuite-adapter/src/query.ts:96]: It's on purpose, `isTypeMatch` of `notQuery(query)` returns false only if `query` matches all the instances of `typeName`, i.e., `query.areAllObjectsMatch` is true.

**Pull request merged**